Traceback (most recent call last):
  File "/home/lorenzf/git/data-science-practical-approach/venv/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/lorenzf/git/data-science-practical-approach/venv/lib/python3.8/site-packages/nbclient/client.py", line 1117, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/lorenzf/git/data-science-practical-approach/venv/lib/python3.8/site-packages/nbclient/util.py", line 78, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/lorenzf/git/data-science-practical-approach/venv/lib/python3.8/site-packages/nbclient/util.py", line 57, in just_run
    return loop.run_until_complete(coro)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/lorenzf/git/data-science-practical-approach/venv/lib/python3.8/site-packages/nbclient/client.py", line 558, in async_execute
    await self.async_execute_cell(
  File "/home/lorenzf/git/data-science-practical-approach/venv/lib/python3.8/site-packages/nbclient/client.py", line 862, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/home/lorenzf/git/data-science-practical-approach/venv/lib/python3.8/site-packages/nbclient/client.py", line 765, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
geo_rating_df.pivot(index='alcohol', columns='smoking_area', values='rating')
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
[0;32m/tmp/ipykernel_20513/1351770208.py[0m in [0;36m<module>[0;34m[0m
[0;32m----> 1[0;31m [0mgeo_rating_df[0m[0;34m.[0m[0mpivot[0m[0;34m([0m[0mindex[0m[0;34m=[0m[0;34m'alcohol'[0m[0;34m,[0m [0mcolumns[0m[0;34m=[0m[0;34m'smoking_area'[0m[0;34m,[0m [0mvalues[0m[0;34m=[0m[0;34m'rating'[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m
[0;32m~/git/data-science-practical-approach/venv/lib/python3.8/site-packages/pandas/core/frame.py[0m in [0;36mpivot[0;34m(self, index, columns, values)[0m
[1;32m   7791[0m         [0;32mfrom[0m [0mpandas[0m[0;34m.[0m[0mcore[0m[0;34m.[0m[0mreshape[0m[0;34m.[0m[0mpivot[0m [0;32mimport[0m [0mpivot[0m[0;34m[0m[0;34m[0m[0m
[1;32m   7792[0m [0;34m[0m[0m
[0;32m-> 7793[0;31m         [0;32mreturn[0m [0mpivot[0m[0;34m([0m[0mself[0m[0;34m,[0m [0mindex[0m[0;34m=[0m[0mindex[0m[0;34m,[0m [0mcolumns[0m[0;34m=[0m[0mcolumns[0m[0;34m,[0m [0mvalues[0m[0;34m=[0m[0mvalues[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m   7794[0m [0;34m[0m[0m
[1;32m   7795[0m     _shared_docs[

[0;32m~/git/data-science-practical-approach/venv/lib/python3.8/site-packages/pandas/core/reshape/pivot.py[0m in [0;36mpivot[0;34m(data, index, columns, values)[0m
[1;32m    515[0m         [0;32melse[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m    516[0m             [0mindexed[0m [0;34m=[0m [0mdata[0m[0;34m.[0m[0m_constructor_sliced[0m[0;34m([0m[0mdata[0m[0;34m[[0m[0mvalues[0m[0;34m][0m[0;34m.[0m[0m_values[0m[0;34m,[0m [0mindex[0m[0;34m=[0m[0mmultiindex[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 517[0;31m     [0;32mreturn[0m [0mindexed[0m[0;34m.[0m[0munstack[0m[0;34m([0m[0mcolumns_listlike[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    518[0m [0;34m[0m[0m
[1;32m    519[0m [0;34m[0m[0m

[0;32m~/git/data-science-practical-approach/venv/lib/python3.8/site-packages/pandas/core/series.py[0m in [0;36munstack[0;34m(self, level, fill_value)[0m
[1;32m   4079[0m         [0;32mfrom[0m [0mpandas[0m[0;34m.[0m[0mcore[0m[0;34m.[0m[0mreshape[0m[0;34m.[0m[0mreshape[0m [0;32mimport[0m [0munstack[0m[0;34m[0m[0;34m[0m[0m
[1;32m   4080[0m [0;34m[0m[0m
[0;32m-> 4081[0;31m         [0;32mreturn[0m [0munstack[0m[0;34m([0m[0mself[0m[0;34m,[0m [0mlevel[0m[0;34m,[0m [0mfill_value[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m   4082[0m [0;34m[0m[0m
[1;32m   4083[0m     [0;31m# ----------------------------------------------------------------------[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/git/data-science-practical-approach/venv/lib/python3.8/site-packages/pandas/core/reshape/reshape.py[0m in [0;36munstack[0;34m(obj, level, fill_value)[0m
[1;32m    458[0m         [0;32mif[0m [0mis_1d_only_ea_dtype[0m[0;34m([0m[0mobj[0m[0;34m.[0m[0mdtype[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m    459[0m             [0;32mreturn[0m [0m_unstack_extension_series[0m[0;34m([0m[0mobj[0m[0;34m,[0m [0mlevel[0m[0;34m,[0m [0mfill_value[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 460[0;31m         unstacker = _Unstacker(
[0m[1;32m    461[0m             [0mobj[0m[0;34m.[0m[0mindex[0m[0;34m,[0m [0mlevel[0m[0;34m=[0m[0mlevel[0m[0;34m,[0m [0mconstructor[0m[0;34m=[0m[0mobj[0m[0;34m.[0m[0m_constructor_expanddim[0m[0;34m[0m[0;34m[0m[0m
[1;32m    462[0m         )

[0;32m~/git/data-science-practical-approach/venv/lib/python3.8/site-packages/pandas/core/reshape/reshape.py[0m in [0;36m__init__[0;34m(self, index, level, constructor)[0m
[1;32m    131[0m             [0;32mraise[0m [0mValueError[0m[0;34m([0m[0;34m"Unstacked DataFrame is too big, causing int32 overflow"[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m    132[0m [0;34m[0m[0m
[0;32m--> 133[0;31m         [0mself[0m[0;34m.[0m[0m_make_selectors[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    134[0m [0;34m[0m[0m
[1;32m    135[0m     [0;34m@[0m[0mcache_readonly[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/git/data-science-practical-approach/venv/lib/python3.8/site-packages/pandas/core/reshape/reshape.py[0m in [0;36m_make_selectors[0;34m(self)[0m
[1;32m    183[0m [0;34m[0m[0m
[1;32m    184[0m         [0;32mif[0m [0mmask[0m[0;34m.[0m[0msum[0m[0;34m([0m[0;34m)[0m [0;34m<[0m [0mlen[0m[0;34m([0m[0mself[0m[0;34m.[0m[0mindex[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 185[0;31m             [0;32mraise[0m [0mValueError[0m[0;34m([0m[0;34m"Index contains duplicate entries, cannot reshape"[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    186[0m [0;34m[0m[0m
[1;32m    187[0m         [0mself[0m[0;34m.[0m[0mgroup_index[0m [0;34m=[0m [0mcomp_index[0m[0;34m[0m[0;34m[0m[0m

[0;31mValueError[0m: Index contains duplicate entries, cannot reshape
ValueError: Index contains duplicate entries, cannot reshape

